### DISCUSSION TITLE
ESB Networks Ireland

### DISCUSSION THREAD

**Discussion by DamianFlynn** [UtilitechAS/amsreader-firmware#473] [Q&A]
Hi Guys. 

I have recently had a Kamstrup DK-8660 installed and wish to get the energy data from the meter. I have a bunch of ESP chips from a xlights project that are spare and found this project and the Power-K product. 

Has anyone tried this with the Irish installations and had any luck. I’m willing to give it a good try and just need some pointers. 

Thx
Damian

**Comment by ArnieO** [comment]
Hello Ireland!
Is your meter a Kamstrup Omnipower? Maybe share a photo of the meter here?

**Comment by DamianFlynn** [comment]
![IMG_9761](https://user-images.githubusercontent.com/6288143/223203567-3c71433a-d91b-4f3b-bf3c-64d24302b9aa.jpg)
![IMG_9762](https://user-images.githubusercontent.com/6288143/223203598-656a4da4-794f-43e8-ad9d-6b3a5d4f162a.jpg)

**Reply by ArnieO** [reply]
Link does not work, please try again!

**Reply by DamianFlynn** [reply]
wrong format, fixed now

**Reply by ArnieO** [reply]
Ok, this is a Kamstup Omnipower. "DK-8660" is not the model number, I believe it is the postal code at the location of Kamstrup in Denmark. 

All Kamstrup Omnipower meters shall be configured or configurable to "Data push" mode, which is necessary for this firmware to work. Can you find technical information on the website of your grid company?

PS: A [Pow-K](https://amsleser.no/blog/post/17-video-how-to-install-and-configure-your-pow-k) can be inserted if you remove the lid on top of the meter. But if I understand you correctly, you want to make your own solution?

**Reply by ArnieO** [reply]
![image](https://user-images.githubusercontent.com/10295178/223206530-f6359750-fe74-44da-b50c-95607cc7cc85.png)

**Reply by DamianFlynn** [reply]
That message for ESB is the same one we got in recent days via email telling us the portal
Is now ready. I am waiting on getting account reset to login. 

I am happy to go either way. The module
Would be cleaner install. But to get going I can program a wemos D1 or node mcu. 

I’m due in Oslo at the end of May for NIC. Or can have the unit ordered for delivery here or to the office in Oslo.

**Reply by DamianFlynn** [reply]
Reading the WiKi I require an MBus module for the esp to communicate with the meter so in this case I might be better getting the pow-k module 

I also read that we might need some decryption key, what would you advise to check that this would not be a total waste of cash? (Sorry blind question I understand)

**Reply by ArnieO** [reply]
> That message for ESB is the same one we got in recent days via email telling us the portal Is now ready. I am waiting on getting account reset to login.
> 
> I am happy to go either way. The module Would be cleaner install. But to get going I can program a wemos D1 or node mcu.
> 
> I’m due in Oslo at the end of May for NIC. Or can have the unit ordered for delivery here or to the office in Oslo.

It is feasible to get the firmware running on a Wemos D1 mini, but there are some pitfalls you need to avoid. Also, some D1 minis have insufficient decoupling and might be unstable due to the combination of relatively heavy exploitation of Wifi and also reliance on the ESP auto sleep functions due to limitations in current draw allowed from the power meter. So you should beef up the Wemos with more capacitance on the 3,3V. The Pow-K uses a 1F supercapacitor to ensure stable voltage (surely over dimensioned - but it works well). 

If you want to try first with a Wemos you need to connect to a 6-pin connector sitting behind the lid on top of your photos. Pinout: 
![image](https://user-images.githubusercontent.com/10295178/223357619-b38b9a76-f6de-4667-af98-50822cde31d2.png)
![image](https://user-images.githubusercontent.com/10295178/223357661-71d376bd-8e0e-40d9-96fc-15e291e0895a.png)

If you just do that to test before ordering a Pow-K (which is a good idea if you're uncertain regarding encryption keys etc),
just run it from a USB power, it simplifies things and should hopefully be stable (ref above comments). 

If you want to go for a Pow-K you can order it from our website [amsleser.no](https://amsleser.no/hjem/1-pow-k-kamstrup.html) for delivery to Norway, or from [our store on Lectronz](https://lectronz.com/stores/amsleser/) for delivery to Ireland (Lectronz handles the VAT to EU countries).

Your grid company will provide the encryption keys for your meter. In most countries the keys are sent to you when you request activation of the port (it is by default inactive). 

You mention the Mbus module, I assume you mean this one: https://www.kamstrup.com/en-us/electricity-solutions/smart-electricity-meters/modules/han-nve

My advise is to avoid it, as we have seen some cases where it has created issues. Kamstrup meters fitted with this module can use the Pow-U, but the module delivers a very limited amount of power on the M-bus, and in some cases not enough to power the Pow-U (it then needs external power via USB).

**Reply by ArnieO** [reply]
This video shows how you install and configure a Pow-K on your Kamstrup meter: https://amsleser.no/blog/post/17-video-how-to-install-and-configure-your-pow-k

**Comment by DamianFlynn** [comment]
I found the following document which is
Quite detailed. https://www.esbnetworks.ie/docs/default-source/publications/smart-metering-september-2022-dpia-final-.pdf?

**Reply by ArnieO** [reply]
Yeah, but the subject here is the centralized data collection system used by the grid company, not the HAN port on your meter.

**Comment by DamianFlynn** [comment]
I grabbed a NodeMCU, installed Platform.IO and compiled the application, and have it uploaded ready for testing. I plan to place the device in the meter box with USB battery to power the test, then connect GPIO12 (D6) to the meter Pin5 and GND to meter Pin6.

My assumption is that the HAN should change from red, but I have no idea if the port is enabled? I think i read in some places that this might not be enabled and i need to call someone! god know who in ireland...

then the second issue is that the data if it is streamed maybe encrypted...

How will i know if this is the case? HAN will change from red to different colour, but no consumption data?

thank you, Damian

**Reply by DamianFlynn** [reply]
found these gold mines of information

https://github.com/UtilitechAS/amsreader-firmware/issues/115

**Reply by ArnieO** [reply]
https://www.esbnetworks.ie/existing-connections/meters-and-readings/my-smart-data

**Reply by DamianFlynn** [reply]
I read that note, but I did not extract any details to suggest what maybe possible physically. My assumption is nothing and we are to live with this daily export workaround

**Reply by ArnieO** [reply]
Not quite sure what you mean now, but the HAN port on these meters is by default not activated. I assume you need to create account and log in there to request activation.
